core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/hashJoin/HashJoinOpExec.scala

```diff
@@ -47,14 +47,14 @@ class HashJoinOpExec[K](val opDesc: HashJoinOpDesc[K]) extends OperatorExecutor
 
   override def processTexeraTuple(
       tuple: Either[Tuple, InputExhausted],
-      input: Int
+      input: OperatorIdentifier
   ): Iterator[Tuple] = {
     tuple match {
       case Left(t) =>
         // The operatorInfo() in HashJoinOpDesc has a inputPorts list. In that the
         // small input port comes first. So, it is assigned the inputNum 0. Similarly
         // the large input is assigned the inputNum 1.
-        if (input == 0) {
+        if (opDesc.opExecConfig.getInputNum(input) == 0) {
```

Comment:
I suggest we remove all usage of input numbers. this `getInputNum` function should be removed